src/quacc/atoms/skzcam.py

```diff
@@ -13,12 +13,489 @@
+    element_info
+        A dictionary with elements as keys which gives the (1) number of core electrons as 'core', (2) basis set as 'basis', (3) effective core potential as 'ecp', (4) resolution-of-identity/density-fitting auxiliary basis set for DFT/HF calculations as 'ri_scf_basis' and (5) resolution-of-identity/density-fitting for correlated wave-function methods as 'ri_cwft_basis'.
+    pal_nprocs_block
+        A dictionary with the number of processors for the PAL block as 'nprocs' and the maximum memory-per-core in megabytes blocks as 'maxcore'.
+    method_block
+        A dictionary that contains the method block for the ORCA input file.
+    scf_block
+        A dictionary that contains the SCF block for the ORCA input file.
+    ecp_info
+        A dictionary with the ECP data (in ORCA format) for the cations in the ECP region.
+    include_cp
+        If True, the coords strings will include the counterpoise correction for the adsorbate and slab.
+    multiplicity
+        The multiplicity of the adsorbate-slab complex, adsorbate and slab respectively.
+
+    Returns
+    -------
+    Tuple[str,str,str]
+        The coordinates block for the adsorbate-slab complex, the adsorbate, and the slab.
+    """
+
+    # First generate the preamble block
+    preamble_block = generate_orca_input_preamble(
+        embedded_adsorbed_cluster,
+        quantum_cluster_indices,
+        element_info,
+        pal_nprocs_block,
+        method_block,
+        scf_block,
+    )
+
+    # Generate the coords block
+    ad_slab_coords, ad_coords, slab_coords = generate_coords_block(
+        embedded_adsorbed_cluster,
+        quantum_cluster_indices,
+        ecp_region_indices,
+        ecp_info,
+        include_cp,
+        multiplicity,
+    )
+
+    # Combine the blocks
+    ad_slab_block = preamble_block + ad_slab_coords
+    ad_block = preamble_block + ad_coords
+    slab_block = preamble_block + slab_coords
+
+    return ad_slab_block, ad_block, slab_block
+
+
+def create_atom_coord_string(
+    atom: Atom,
+    ghost_atom: bool = False,
+    atom_ecp_info: str | None = None,
+    pc_charge: float | None = None,
+) -> str:
+    """
+    Creates a string containing the Atom symbol and coordinates in the ORCA input file format, with additional information for atoms in the ECP region as well as ghost atoms.
+
+    Parameters
+    ----------
+    atom
+        The ASE Atom (not Atoms) object containing the atomic coordinates.
+    atom_ecp_info
+        If not None, then assume this is an atom in the ECP region and add the ECP info.
+    ghost_atom
+        If True, then the atom is a ghost atom.
+    pc_charge
+        The point charge value for the ECP region atom.
+
+    Returns
+    -------
+    str
+        The atom symbol and coordinates in the ORCA input file format.
+    """
+
+    # If ecp_info is not None and ghost_atom is True, raise an error
+    if atom_ecp_info is not None and ghost_atom:
```

Comment:
Can likely (?) just do `if atom_ecp_info and ghost_atom` unless `atom_ecp_info` can be `""`. Similar comment in the line below.